FreeSurfer/FreeSurferPipelineNHP.sh

```diff
@@ -0,0 +1,1219 @@
+			log_Err_Abort "tkregister command failed with return_code: ${return_code}"
+		fi
+
+		log_Msg "...Clean up"
+		rm deleteme.dat
+		rm Q.lta
+
+		popd
+	fi
+
+	# ----------------------------------------------------------------------
+	log_Msg "Creating white surface files in rawavg space"
+	# ----------------------------------------------------------------------
+
+	pushd ${mridir}
+	
+	export SUBJECTS_DIR="$SubjectDIR"
+	
+	reg=$mridir/transforms/orig2rawavg.dat
+	# generate registration between conformed and hires based on headers
+	# Note that the convention of tkregister2 is that the resulting $reg is the registration
+	# matrix that maps from the "--targ" space into the "--mov" space. 
+	
+	tkregister2 --mov ${mridir}/rawavg.mgz --targ ${mridir}/orig.mgz --noedit --regheader --reg $reg
+	
+	#The ?h.white.deformed surfaces are used in FreeSurfer BBR registrations for fMRI and diffusion and have been moved into the HCP's T1w space so that BBR produces a transformation containing only the minor adjustment to the registration.
+	#The ?h.pial.deformed surfaces are used for USPIO fMRI registration - TH 2024
+	for hemi in lh rh; do
+		for surf in white pial; do
+			mri_surf2surf --s ${SubjectID} --sval-xyz "$surf" --reg "$reg" --tval-xyz ${mridir}/rawavg.mgz --tval "$surf".deformed --surfreg "$surf" --hemi "$hemi"
+			return_code=$?
+			if [ "${return_code}" != "0" ] ; then
+				log_Err_Abort "mri_surf2surf command for $hemi hemisphere $surf failed with return_code: ${return_code}"
+			fi
+		done
+	done
+	
+	popd
+	
+	# ----------------------------------------------------------------------
+	log_Msg "Generating QC file"
+	# ----------------------------------------------------------------------
+
+	make_t1w_hires_nifti_file "${mridir}"
+
+	if [ "${T2wImage}" != "NONE" ] ; then
+
+		make_t2w_hires_nifti_file "${mridir}"
+
+		make_t1wxt2w_qc_file "${mridir}"
+	fi
+fi
+# ----------------------------------------------------------------------
+log_Msg "Completing main functionality"
+# ----------------------------------------------------------------------
+
+
+
+
+# Global processing - everything above here should be in a function
+
+g_script_name=$(basename "${0}")
+
+
+# ------------------------------------------------------------------------------
+#  Check that HCPPIPEDIR is defined and Load Function Libraries
+# ------------------------------------------------------------------------------
+
+if [ -z "${HCPPIPEDIR}" ] ; then
+	echo "${g_script_name}: ABORTING: HCPPIPEDIR environment variable must be set"
+	exit 1
+fi
+
+source "${HCPPIPEDIR}/global/scripts/debug.shlib" "$@"         # Debugging functions; also sources log.shlib
+source ${HCPPIPEDIR}/global/scripts/opts.shlib                 # Command line option functions
+source ${HCPPIPEDIR}/global/scripts/processingmodecheck.shlib  # Check processing mode requirements
+
+opts_ShowVersionIfRequested $@
+
+if opts_CheckForHelpRequest $@; then
+	show_usage
+	exit 0
+fi
+
+${HCPPIPEDIR}/show_version
```

Comment:
This is all cruft from the "main function" style of old `opts.shlib`, the start of the script uses newopts instead.